api/v1alpha3/virtualmachine_storage_types.go

```diff
@@ -66,12 +66,48 @@ type InstanceVolumeClaimVolumeSource struct {
 	Size resource.Quantity `json:"size"`
 }
 
+// +kubebuilder:validation:Enum=Classic;Managed
+
+// VirtualMachineVolumeType describes the type of a VirtualMachine volume.
+type VirtualMachineVolumeType string
+
+const (
+	// VirtualMachineStorageDiskTypeClassic describes a classic virtual disk,
+	// such as the boot disk for a VirtualMachine deployed from a VM Image of
+	// type OVF.
+	VirtualMachineStorageDiskTypeClassic VirtualMachineVolumeType = "Classic"
+
+	// VirtualMachineStorageDiskTypeManaged describes a managed virtual disk,
+	// such as persistent volumes.
+	VirtualMachineStorageDiskTypeManaged VirtualMachineVolumeType = "Managed"
+)
+
 // VirtualMachineVolumeStatus defines the observed state of a
 // VirtualMachineVolume instance.
 type VirtualMachineVolumeStatus struct {
 	// Name is the name of the attached volume.
 	Name string `json:"name"`
 
+	// +kubebuilder:default=Managed
+
+	// Type is the type of the attached volume.
+	Type VirtualMachineVolumeType `json:"type"`
```

Comment:
@bryanv Do you know if we can safely make this non-optional? Since it has a default value I would think so, but I am not sure without checking, and we don't have any in-repo examples for setting up a test for an existing status already recorded without a type. 

This is all complicated of course by the fact that volume status info is now handled in two distinct locations.

Reply:
I think we can be OK - on the initial up version conversion - when we don't have the the curver saved in the VM annotation - just assume everything is Managed since that is what everything was before. If we get it wrong, it will be fixed up on a subsequent reconcile.